src/rules/tsUtils.ts

```diff
@@ -108,6 +108,128 @@ export const isStringNode = <V extends string>(
 export const getStringValue = <S extends string>(node: StringNode<S>): S =>
   isTemplateLiteral(node) ? node.quasis[0].value.raw : node.value;
 
+/**
+ * Represents a `MemberExpression` with a "known" `property`.
+ */
+interface KnownMemberExpression<Name extends string = string>
+  extends TSESTree.MemberExpression {
+  property: AccessorNode<Name>;
+}
+
+/**
+ * An `Identifier` with a known `name` value - i.e `expect`.
+ */
+interface KnownIdentifier<Name extends string> extends TSESTree.Identifier {
+  name: Name;
+}
+
+/**
+ * Checks if the given `node` is an `Identifier`.
+ *
+ * If a `name` is provided, & the `node` is an `Identifier`,
+ * the `name` will be compared to that of the `identifier`.
+ *
+ * @param {Node} node
+ * @param {V?} name
+ *
+ * @return {node is KnownIdentifier<Name>}
+ *
+ * @template V
+ */
+const isIdentifier = <V extends string>(
+  node: TSESTree.Node,
+  name?: V,
+): node is KnownIdentifier<V> =>
+  node.type === AST_NODE_TYPES.Identifier &&
+  (name === undefined || node.name === name);
+
+/**
+ * Checks if the given `node` is a "supported accessor".
+ *
+ * This means that it's a node can be used to access properties,
+ * and who's "value" can be statically determined.
+ *
+ * `MemberExpression` nodes most commonly contain accessors,
+ * but it's possible for other nodes to contain them.
+ *
+ * If a `value` is provided & the `node` is an `AccessorNode`,
+ * the `value` will be compared to that of the `AccessorNode`.
+ *
+ * Note that `value` here refers to the normalised value.
+ * The property that holds the value is not always called `name`.
+ *
+ * @param {Node} node
+ * @param {V?} value
+ *
+ * @return {node is AccessorNode<V>}
+ *
+ * @template V
+ */
+export const isSupportedAccessor = <V extends string>(
+  node: TSESTree.Node,
+  value?: V,
+): node is AccessorNode<V> =>
+  isIdentifier(node, value) || isStringNode(node, value);
+
+/**
+ * Gets the value of the given `AccessorNode`,
+ * account for the different node types.
+ *
+ * @param {AccessorNode<S>} accessor
+ *
+ * @return {S}
+ *
+ * @template S
+ */
+export const getAccessorValue = <S extends string = string>(
+  accessor: AccessorNode<S>,
+): S =>
+  /* istanbul ignore next */
```

Comment:
This is b/c we have no rules that use this on an `Identifier` so coverage complains, *but* `no-large-snapshots` uses `isSupportedAccessor` on an `Identifier`, meaning `AccessorNode` needs to be a union, and so TypeScript requires us to handle both branches.

It's a catch-22 :)

Reply:
you solved this another way? Seems to be gone

Reply:
I solved it by using it - this was never needed in this branch, but slipped in via the rebase

Reply:
Sorry thought this was #363 - my tests locally still require this for this PR?

Reply:
this line doesn't exist anymore